library/core/tests/num/int_macros.rs

```diff
@@ -288,6 +288,7 @@ macro_rules! int_module {
                 assert_eq!(r.saturating_pow(2), 4 as $T);
                 assert_eq!(r.saturating_pow(3), -8 as $T);
                 assert_eq!(r.saturating_pow(0), 1 as $T);
+                assert_eq!(2i64.checked_pow(64), None); // issue #120537
```

Comment:
These tests are expanded for all integer types `$T`, so it would be nice to keep it testing each type. In this signed case, above where `r = 2 as $T` we can check exponent `<$T>::BITS - 1` produces `None`, and with `r = -2` we can check that `BITS - 1` is `Some(<$T>::MIN)` and `BITS` gets `None`.